selftests/functional/basic.py

```diff
@@ -99,9 +99,12 @@ def test(self):
 """
 
 EXEC_ENV_VARIABLE_TEST = """#!/bin/bash
-if [[ -n "${TEST_ENV}"  ]]; then
+if [[ -n "${TEST_ENV}" ]]; then
```

Comment:
It looks like a spurious space is removed here in the 4th patch, better to fold that fix into patch 3 where the code is introduced.